code/game/objects/structures/inflatable.dm

```diff
@@ -0,0 +1,328 @@
+
+/obj/structure/inflatable/door/deflate(var/violent=0)
+	playsound(loc, 'sound/machines/hiss.ogg', 75, 1)
+	if(violent)
+		visible_message("<span class='danger'>[src] rapidly deflates!</span>")
+		var/obj/item/inflatable/door/torn/R = new /obj/item/inflatable/door/torn(loc)
+		src.transfer_fingerprints_to(R)
+		qdel(src)
+	else
+		visible_message("<span class='danger'>[src] slowly deflates.</span>")
+		spawn(50)
+			var/obj/item/inflatable/door/R = new /obj/item/inflatable/door(loc)
+			src.transfer_fingerprints_to(R)
+			qdel(src)
+	air_update_turf(1)
+
+/obj/item/inflatable/torn
+	name = "torn inflatable wall"
+	desc = "A folded membrane which rapidly expands into a large cubical shape on activation. It is too torn to be usable."
+	icon = 'icons/obj/inflatable.dmi'
+	icon_state = "folded_wall_torn"
+
+/obj/item/inflatable/torn/attack_self(mob/user)
+	user << "<span class='warning'>The inflatable wall is too torn to be inflated, fix it with something!</span>"
+	add_fingerprint(user)
+
+/obj/item/inflatable/torn/attackby(obj/item/I, mob/user)
+	if(istype(I, /obj/item/stack/ducttape))
+		var/obj/item/stack/ducttape/T = I
+		if(T.amount < 2)
+			user << "<span class='danger'>There is not enough tape!</span>"
+			return
+		user << "<span class='notice'>You begin fixing the [src]!</span>"
+		playsound(user, 'sound/items/ducttape1.ogg', 50, 1)
+		if(do_mob(user, src, 20))
+			user << "<span class='notice'>You fix the [src] using the ducttape!</span>"
+			T.use(2)
+			new /obj/item/inflatable(user.loc)
+			qdel(src)
+
+/obj/item/inflatable/door/torn
+	name = "torn inflatable door"
+	desc = "A folded membrane which rapidly expands into a simple door on activation. It is too torn to be usable."
+	icon = 'icons/obj/inflatable.dmi'
+	icon_state = "folded_door_torn"
+
+/obj/item/inflatable/door/torn/attack_self(mob/user)
+	user << "<span class='warning'>The inflatable door is too torn to be inflated, fix it with something!</span>"
+	add_fingerprint(user)
+
+
+/obj/item/inflatable/door/torn/attackby(obj/item/I, mob/user)
+	if(istype(I, /obj/item/stack/ducttape))
+		var/obj/item/stack/ducttape/T = I
+		if(T.amount < 2)
+			user << "<span class='danger'>There is not enough tape!</span>"
+			return
+		user << "<span class='notice'>You begin fixing the [src]!</span>"
+		playsound(user, 'sound/items/ducttape1.ogg', 50, 1)
+		if(do_mob(user, src, 25))
+			user << "<span class='notice'>You fix the [src] using the ducttape!</span>"
+			T.use(2)
+			new /obj/item/inflatable/door(user.loc)
+			qdel(src)
+
+/obj/item/weapon/storage/inflatable
+	name = "inflatable barrier box"
+	desc = "Contains inflatable walls and doors."
+	icon_state = "inf"
+	item_state = "syringe_kit"
+	max_combined_w_class = 21
+	w_class = 3
+
+/obj/item/weapon/storage/inflatable/New()
+	..()
+	for(var/i = 0, i < 8, i ++)
+		new /obj/item/inflatable/door(src)
+	for(var/i = 0, i < 16, i ++)
+		new /obj/item/inflatable(src)
+
+/obj/item/inflatable/suicide_act(mob/living/user)
```

Comment:
suicide acts must return the kind of damage they do as a define, like return BRUTELOSS